src/System.IO.StreamSourceGeneration/src/StreamSourceGen.cs

```diff
@@ -0,0 +1,177 @@
+            .ForAttributeWithMetadataName(
+                StreamBoilerplateAttributeFullName,
+                predicate: (node, _) => node is ClassDeclarationSyntax c && c.Modifiers.Any(SyntaxKind.PartialKeyword),
+                transform: (context, _) => (ClassDeclarationSyntax)context.TargetNode);
+
+        IncrementalValueProvider<(Compilation, ImmutableArray<ClassDeclarationSyntax>)> compilationAndClasses =
+            context.CompilationProvider.Combine(classDeclarations.Collect());
+
+        context.RegisterSourceOutput(compilationAndClasses, (spc, source) => Execute(source.Item1, source.Item2, spc));
+    }
+
+    private static void Execute(Compilation compilation, ImmutableArray<ClassDeclarationSyntax> classes, SourceProductionContext context)
+    {
+        if (classes.IsDefaultOrEmpty)
+        {
+            return;
+        }
+
+#if LAUNCH_DEBUGGER
+        if (!Diagnostics.Debugger.IsAttached)
+        {
+            Diagnostics.Debugger.Launch();
+        }
+#endif
+
+        List<StreamTypeInfo>? classesWithGenerationOptions = Helpers.GetClassesWithGenerationOptions(compilation, classes, context.CancellationToken);
+
+        if (classesWithGenerationOptions == null)
+        {
+            return;
+        }
+
+        StringBuilder sb = new();
+
+        foreach (StreamTypeInfo streamTypeInfo in classesWithGenerationOptions)
+        {
+            sb.Clear();
+            Emit(sb, streamTypeInfo);
+
+            INamedTypeSymbol typeSymbol = streamTypeInfo.TypeSymbol;
+            string hintName = $"{typeSymbol.ContainingNamespace}.{typeSymbol.Name}.Boilerplate.g.cs";
+            context.AddSource(hintName, sb.ToString());
+        }
+    }
+
+    private static void Emit(StringBuilder sb, StreamTypeInfo streamTypeInfo)
+    {
+        sb.Append($@"// <auto-generated/>
+
+#nullable enable
+
+{StreamBoilerplateConstants.UsingDirectives}
+
+namespace {streamTypeInfo.TypeSymbol.ContainingNamespace}
+{{");
+        sb.Append($@"
+    partial class {streamTypeInfo.TypeSymbol.Name}
+    {{");
+
+        HashSet<string> overriddenMembers = streamTypeInfo.OverriddenMembers;
+
+        // Read*/Write*/Seek can hint that a stream CanRead/Write/Seek => true but not vice versa.
+        bool canRead = streamTypeInfo.ReadInfo != null;
+        bool canWrite = streamTypeInfo.WriteInfo != null;
+        bool canSeek = overriddenMembers.Contains(StreamMembersConstants.Seek);
+
+        foreach (BoilerplateCandidateInfo candidateInfo in BoilerplateCandidateInfo.CandidatesList)
+        {
+            string candidateName = candidateInfo.Name;
+
+            if (overriddenMembers.Contains(candidateName))
+            {
+                continue;
+            }
+
+            StreamOperationKind candidateOperationKind = candidateInfo.OperationKind;
+
+            if (candidateOperationKind >= StreamOperationKind.Read &&
+                candidateOperationKind <= StreamOperationKind.WriteAsync)
```

Comment:
The range on this feels pretty arbitrary, and I had to look it up.

I don't necessarily have a replacement strategy off the top of my head; but I just thought I'd point it out.